kitsune/tidings/tasks.py

```diff
@@ -1,13 +1,55 @@
+from collections.abc import Sequence
+
 from celery import shared_task
+from django.contrib.auth import get_user_model
 
 from kitsune.tidings.models import Watch
+from kitsune.tidings.utils import get_class
 
 
 @shared_task
-def claim_watches(user):
-    """Attach any anonymous watches having a user's email to that user.
-
+def claim_watches(user_id):
+    """
+    Attach any anonymous watches having a user's email to that user.
     Call this from your user registration process if you like.
-
     """
+    user = get_user_model().objects.get(id=user_id)
     Watch.objects.filter(email=user.email).update(email=None, user=user)
+
+
+@shared_task
+def fire(
```

Comment:
I have mixed feelings about this approach. We have this task which is called by a `classmethod` that gets an instance sometimes and the task constructs the object. Wouldn't be simpler to just move the task in each subclass of Event? The main event could have a synchronous `run`, `send_emails` or any other suitable name that each `fire` method from the subclass would run asynchronously

Reply:
There's a lot to discuss here, so probably better to discuss in a video meeting, but I have three thoughts.
- If we move the task to each subclass of `Event`, those tasks would no longer be auto-discoverable by https://github.com/mozilla/kitsune/blob/2b40d5841ceeb6bd2f8108ebfd3decbec97a168f/kitsune/celery.py#L10.
- The `fire_async` `classmethod` serves the purpose of translating the incoming arguments that reference Python objects, like `instance` and `exclude`, into integers and strings and then calling the Celery task with those JSON-friendly arguments.
- If you don't like this general approach of having a single Celery task in tidings, there is a [version of this PR (an earlier commit)](https://github.com/mozilla/kitsune/pull/5169/commits/11f050149a0d22dee56f48a136742a44c4022426) that creates separate Celery tasks for each Event sub-class.